File: examples/ios-demo/SDKDemo/RecordPenDemo/Utils/LocalizationHelper.m
#import "LocalizationHelper.h"

@implementation LocalizationHelper

+ (NSString *)localizedString:(NSString *)key {
    NSString *result = NSLocalizedString(key, nil);
    
#ifdef DEBUG
    // If the returned string is the same as the key, it means no localized string was found
    if ([result isEqualToString:key]) {
        NSLog(@"⚠️ Localization missing for key: %@", key);
        NSLog(@"Current language: %@", [self currentLanguage]);
        NSLog(@"Bundle: %@", [NSBundle mainBundle]);
        NSLog(@"Localizations: %@", [NSBundle mainBundle].localizations);
        
        // Check if Localizable.strings file exists
        NSString *path = [[NSBundle mainBundle] pathForResource:@"Localizable" ofType:@"strings" inDirectory:nil forLocalization:[self currentLanguage]];
        NSLog(@"Localizable.strings path: %@", path);
    }
#endif
    
    return result;
}

+ (NSString *)localizedStringWithFormat:(NSString *)key, ... {
    va_list args;
    va_start(args, key);
    NSString *format = NSLocalizedString(key, nil);
    
#ifdef DEBUG
    // If the returned string is the same as the key, it means no localized string was found
    if ([format isEqualToString:key]) {
        NSLog(@"⚠️ Localization missing for key: %@", key);
    }
#endif
    
    NSString *result = [[NSString alloc] initWithFormat:format arguments:args];
    va_end(args);
    return result;
}

+ (NSString *)currentLanguage {
    NSString *language = [NSBundle mainBundle].preferredLocalizations.firstObject;
#ifdef DEBUG
    NSLog(@"Current language: %@", language);
#endif
    return language ?: @"en";
}

+ (NSString *)displayNameForLanguage:(NSString *)code {
    NSLocale *locale = [[NSLocale alloc] initWithLocaleIdentifier:code];
    NSString *displayName = [locale displayNameForKey:NSLocaleIdentifier value:code];
    return displayName ?: code;
}

@end 

File: examples/ios-demo/SDKDemo/RecordPenDemo/Utils/LocalizationHelper.swift
import Foundation

@objc public class LocalizationHelper: NSObject {
    @objc public static func localizedString(_ key: String) -> String {
        return NSLocalizedString(key, comment: "")
    }

    @objc public static func localizedString(_ key: String, arguments: CVarArg...) -> String {
        let format = NSLocalizedString(key, comment: "")
        return String(format: format, arguments: arguments)
    }

    // Get current language
    @objc public static var currentLanguage: String {
        return Bundle.main.preferredLocalizations.first ?? "en"
    }

    // Check if it's a Chinese environment
    @objc public static var isChineseLanguage: Bool {
        return currentLanguage.hasPrefix("zh")
    }

    // Get language display name
    @objc public static func displayNameForLanguage(_ code: String) -> String {
        let locale = NSLocale(localeIdentifier: code)
        return locale.displayName(forKey: .identifier, value: code) ?? code
    }
}
